src/app/content/highlights/utils/summaryHighlightsUtils.ts

```diff
@@ -21,6 +22,35 @@ interface DataAdd extends BaseData {
   highlight: Highlight;
 }
 
+const insertHighlightAtIndex = (
+  highlights: HighlightData[],
+  highlight: HighlightData, index: number
+) => {
+    return [
+        ...highlights.slice(0, index),
+        highlight,
+        ...highlights.slice(index),
+    ];
+};
+
+export const insertHighlightInOrder = (prevHighlights: HighlightData[] , newHighlight: HighlightData) => {
+  if (!prevHighlights || !prevHighlights.length) {
+      return [newHighlight];
+  }
```

Comment:
I don't think you have to check for `prevHighlights` because typescript will do this
+ indentaiton